rand_distr/src/zipf.rs

```diff
@@ -75,28 +82,27 @@ where F: Float, Standard: Distribution<F>, OpenClosed01: Distribution<F>
 {
     #[inline]
     fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> F {
-        // This is based on the numpy implementation.
+        // This is based on https://doi.org/10.1007/978-1-4613-8643-8.
         loop {
             let u = rng.sample(OpenClosed01);
-            let v = rng.sample(Standard);
             let x = u.powf(-F::one() / self.a_minus_1).floor();
-
-            if x < F::one() {
-                continue;
-            }
+            debug_assert!(x >= F::one());
```

Comment:
Do we want to do something for `x == f64::INFINITY`?

This can happen because before applying floor the number is infinity, but it should fall back to `f64::MAX`. This is also a rounding issue, no? (I know it is an edge-case)

Reply:
Not sure how `f64::MAX` is better than `f64::INFINITY`?

Reply:
If `x` is `f64::INFINITY`, the code will do some computations and then `continue` the loop. This is biasing the distribution, and it does not happen when `x` is `f64::MAX`.

I see two options:
1) Afford this little bias. 
2) Panic (in debug mode?).

Both options have been discussed in the crate, I just don't know what is the best here.

Reply:
> If `x` is `f64::INFINITY`, the code will do some computations and then `continue` the loop. This is biasing the distribution, and it does not happen when `x` is `f64::MAX`.

I don't think the loop will `continue`. If `x` is `f64::INFINITY`, then `t` will be `1.`, causing the loop to end, returning `x`. I think this is the best we can do? I don't see how mapping infinity to a different value will result in less bias.

Reply:
Maybe I am wrong, but this is what I think happens when `x` is `f64::INFINITY`.
- `t` is `1`.
- Therefore `x * (t - F::one())` is `NaN`
- So `v * x * (t - F::one()) / (self.b - F::one())` is also `NaN`
- Finally, `v * x * (t - F::one()) / (self.b - F::one()) <= t / self.b` is `false`

This is why I think the loop continues. Did I get it wrong?

Reply:
I think you got it right, I missed that there is a `f64::INFINITY * 0.`. 😅 

I see the following options:

1. Return `f64::INFINITY`. This is analog to what would happen with `x = f64::MAX`.
2. `continue` the rejection sampling (current behavior). This is equivalent to cutting of the distribution for `x > f64::MAX`, which is acceptable. However, this *might* result in an infinite loop for pathologically small `a`.
3. Panic.

I would like to avoid panicking, and infinite loops, so option 1 looks best to me.

Reply:
I understand it is the best one can do.

I would strongly encourage a warning in the documentation of this decision. Something like the following
```rust
/// # Remarks
///
/// `Zeta` implements `Distribution<F>`, where `F` represents a `float`. Therefore, infinity is a possible sample. 
/// Moreover, a value of infinity might be returned for two reasons: 
/// (1) it is the best representation in the type `F` of the actual sample.
/// (2) to prevent infinite loops 
```

I know it is long, so feel free to shorten it. This will give more control (and responsibility) to the user, so it is in line with the decision of implementing `Distribution<F>` instead of `Distribution<u64>` :)

Reply:
Well, we would have the same problem with `Distribution<u64>`, where we would have to return `u64::MAX`. It's even trickier, because we would have to pick between returning the unbiased, truncated distribution, or having infinite loops.

I agree that a warning is in order.